myhpi/core/templatetags/minutes_tags.py

```diff
@@ -5,21 +5,23 @@
 
 @register.simple_tag(takes_context=True)
 def next_minutes(context, minutes):
-    all_minutes = list(minutes.get_parent().specific.get_visible_minutes(context.request))
-    try:
-        i = all_minutes.index(minutes)
-        if i > 0:
-            return all_minutes[i - 1]
-    except ValueError:
-        return
+    # This may be None if there is no next minutes.
+    return (
+        minutes.get_parent()
+        .specific.get_visible_minutes(context.request)
+        .exclude(basepage_ptr_id__lte=minutes.basepage_ptr_id, date=minutes.date)
```

Comment:
There may be minutes on the same date. Therefore, I can not filter using greater than. Without excluding the current minute, it might link to the same minute. I do not see a better way here, do you?

Reply:
I see, thats a fair point. But then excluding just by the basepage id should be sufficient, as far as I see it there is no need to exclude the date

Reply:
Okay, I thought about this a little more and I think the reason for all this is the following:

I filter out all minutes older than the current one. This leaves all newer ones and the ones with the same date.  
I cannot just throw away all minutes with the same date as there may be multiple minutes on the same date and I want to correctly enumerate all of them.  
I therefore only throw away all minutes on the same day with an ID less than or equal to the current ID. This leaves only minutes on the same day with a larger ID or minutes on a later date. I basically enumerate minutes on the same day in the order of their IDs. Hence, I need the `exclude` to take the date and the ID.

I hope this explanation helps you and is clearer than my previous attempt.

Reply:
Do you understand my explanation @jeriox? :sweat_smile:

Reply:
yes, thank you! I read the .exclude with an OR syntax, which made no sense to me. But you are of course right as it is actually an AND syntax, so your explanation makes perfect sense. thx for explaining again